src/dotnet-install.ps1

```diff
@@ -858,39 +858,54 @@ function Get-AkaMSDownloadLink([string]$Channel, [string]$Quality, [bool]$Intern
     {
-        try {
-            $akaMsDownloadLink = $Response.Headers.GetValues("Location")[0]
+        #get HTTP response
+        #do not pass credentials as a part of the $akaMsLink and do not apply credentials in the GetHTTPResponse function
+        #otherwise the redirect link would have credentials as well
+        #it would result in applying credentials twice to the resulting link and thus breaking it, and in echoing credentials to the output as a part of redirect link
+        $Response= GetHTTPResponse -Uri $akaMsLink -HeaderOnly $true -DisableRedirect $true -DisableFeedCredential $true
+        Say-Verbose "Received response:`n$Response"
+
+        if ([string]::IsNullOrEmpty($Response)) {
+            Say-Verbose "The link '$akaMsLink' is not valid: failed to get redirect location. The resource is not available."
+            return $null
+        }
 
-            if ([string]::IsNullOrEmpty($akaMsDownloadLink)) {
-                Say-Verbose "The aka.ms link '$akaMsLink' is not valid: failed to get redirect location."
+        #if HTTP code is 301 (Moved Permanently), the redirect link exists
+        if  ($Response.StatusCode -eq 301)
+        {
+            try {
+                $akaMsDownloadLink = $Response.Headers.GetValues("Location")[0]
+
+                if ([string]::IsNullOrEmpty($akaMsDownloadLink)) {
+                    Say-Verbose "The link '$akaMsLink' is not valid: server returned 301 (Moved Permanently), but the headers do not contain the redirect location."
+                    return $null
+                }
+
+                Say-Verbose "The redirect location retrieved: '$akaMsDownloadLink'."
+                # This may yet be a link to another redirection. Attempt to retrieve the page again.
+                $akaMsLink = $akaMsDownloadLink
+                continue
+            }
+            catch {
+                Say-Verbose "The link '$akaMsLink' is not valid: failed to get redirect location."
                 return $null
             }
-
-            Say-Verbose "The redirect location retrieved: '$akaMsDownloadLink'."
-            return $akaMsDownloadLink
         }
-        catch {
-            Say-Verbose "The aka.ms link '$akaMsLink' is not valid: failed to get redirect location."
-            return $null
+        elseif ($Response.StatusCode -eq 200)
+        {
+            # We ended up with a valid URL. This is where the resource should be.
+            return $akaMsDownloadLink
         }
+
+        Say-Verbose "The link '$akaMsLink' is not valid: failed to trace redirections to a valid resource. Last received HTTP code is '$Response.StatusCode'."
+        return $null
     }
-    Say-Verbose "The aka.ms link '$akaMsLink' is not valid: failed to get redirect location. The resourse is not available."
+
+    Say-Warning "Aka.ms links have redirected more than the maximum allowed redirections. This may be caused by a cyclic redirection of aka.ms links."
```

Comment:
`Say-Verbose` probably fits better, as all errors above are reported to `Verbose`

Reply:
Done